frontend/lib/identification/util/card_info_utils.dart

```diff
@@ -22,19 +26,35 @@ extension Hashing on CardInfo {
 }
 
 bool isCardExpired(CardInfo cardInfo) {
-  final expirationDay = _getExpirationDayWithTolerance(cardInfo);
-  return expirationDay != null && expirationDay.isBefore(DateTime.now());
+  final expirationDay = _getExpirationDay(cardInfo);
+  if (expirationDay == null) return false;
+  final now = TZDateTime.from(clock.now(), currentTimezone);
+  return expirationDay.isBefore(TZDateTime(currentTimezone, now.year, now.month, now.day));
```

Comment:
💭 With conditional early returns (i. e. returns in an `if` in the middle of the function), I always put the remaining code in the `else` block. This makes it clear from the indentation, when this code gets executed (and particularly when not). While it is quiet easy to see here, in larger functions it gets hard to spot the `if (...) return ...` in the middle of the code. Sadly, this is actively discouraged by a lint rule in the JS code, and I will propose to drop that rule.

```suggestion
  final expirationDay = _getExpirationDay(cardInfo);
  if (expirationDay == null) {
    return false
  } else {
    final now = TZDateTime.from(clock.now(), currentTimezone);
    return expirationDay.isBefore(TZDateTime(currentTimezone, now.year, now.month, now.day));
  }
```